src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -535,16 +532,25 @@ private void Resize(int newSize, bool forceNewHashCodes)
                 }
             }
 
+            int k = 0;
             for (int i = 0; i < count; i++)
             {
-                if (entries[i].hashCode >= 0)
+                if (_entries[i].hashCode >= 0)
                 {
-                    int bucket = entries[i].hashCode % newSize;
-                    entries[i].next = buckets[bucket];
+                    ref Entry entry = ref _entries[i];
+                    int bucket = entry.hashCode % newSize;
+                    entry.next = buckets[bucket];
                     buckets[bucket] = i;
+                    entries[k] = entry;
```

Comment:
should this be 
```c#
buckets[bucket] = k;
entries[k] = entry;
```
`i` is not relevant to the new arrays.

Reply:
That would break lookup so you might want to run the Dictionary tests against this change.